### ISSUE TITLE
[FEATURE_REQUEST] Allow full offline mode

### ISSUE THREAD

**Issue by aghybris** [Lissy93/dashy#521]
### Is your feature request related to a problem? If so, please describe.

Hi. Thanks for this great dashboard!
Could you please add some complete offline or self-hosted mode. Currently it loads some external resources (e.g: kit.fontawesome.com/0821c65656.js) even if no fontawsome icons are used at all. I use it in the lab where many servers have no connection to public internet. The UI is fine. No resoures are missing but the browser is confused about some failing background requests. Should be easy to omit that.
Thanks. 

### Describe the solution you'd like

A config switch / property to disable requests to external URLs like:
ka-p.fontawesome.com/releases/v5.15.4/webfonts/pro-fa-solid-900-5.0.0.woff2

The dashbord should be perfectly usable without internet connection.

### Priority

Medium (Would be very useful)

**Comment by Lissy93** [comment]
Yes, as someone who really cares about privacy also, I see the appeal of full offline mode. So I will add this to the todo list, although I have already got the up next feature list planned, so this might have to come after that.

Just a note, that currently Dashy should function totally fine when offline, unless you've configured it to use any widgets, themes or icons that were not feasible to include in the bundle.

Font Awesome shouldn't be loaded unless you're using it. Which theme are you using? You can also manually disable it with `enableFontAwesome: false` under `appConfig`

**Comment by aghybris** [comment]
Thanks for the hint. I have now configured enableFontAwesome: false
But still 4 external URLs are loaded. The dashboard looks still great and unchanged with these requests failing.

"appConfig":
"theme": "colorful"
"layout": "vertical"
"iconSize": "small"
"enableFontAwesome": false
"language": "de"

https://kit.fontawesome.com/0821c65656.js
https://ka-p.fontawesome.com/releases/v5.15.4/css/pro-v4-shims.min.css?token=0821c65656
https://ka-p.fontawesome.com/releases/v5.15.4/css/pro-v4-font-face.min.css?token=0821c65656
https://ka-p.fontawesome.com/releases/v5.15.4/webfonts/pro-fa-solid-900-5.0.0.woff2

**Comment by Lissy93** [comment]
Thanks so much for the sponsorship! 💖
It's super kind of you :)

**Comment by Lissy93** [comment]
So I was thinking about this last night, and I think the best option would be for there to be zero external network requests in the default setup, but then if the user chooses to use an icon, widget, theme or feature (like status checking) that requires network connection, then requests will be made. And also for the docs to be really clear about what features make what network requests.

I've submitted a PR, #523 which:
- Fixes the `disableFontAwesome` attribute
- Updated the [privacy docs](https://github.com/Lissy93/dashy/blob/FEATURE/improved-network-request-controls/docs/privacy.md#external-requests), to list every possible network request the app makes when certain (off by default) features are enabled

It should be merged either tonight or tomorrow night.

There's also another ticket: #435, which will add an option to download and cache any external item icons, allowing them to work fully when offline.

I know your idea was to have a toggle switch, which would disable all network requests, but implementing that isn't quite as easy as it sounds. (But I'm open to suggestions, if you know of an elegant method, without service workers or wrapping every request).

Of course, this could also be done very easily from the user's end, either in firewall settings, browser settings, or using an extension like [Work Offline](https://add0n.com/work-offline.html).

**Comment by alexdelprete** [comment]
I was about to open an issue more or less for the same request. So glad you already addressed it @Lissy93.

One thing: could ALL icons, no matter the source, be cached locally? Sometimes external sites are not reachable and the icons fallback to the broken icon. It would be good if they are always available in a local cache, with a TTL maybe so they get refreshed when possible.

Thanks again for your excellent work.

**Comment by jg3** [comment]
It seems greater minds have preceeded me here.  Thank you for already addressing this.   I like @alexdelprete 's idea here to cache icons locally.  In that way everything works beautifully if the user is online or off.

**Comment by denganliang** [comment]
I also have this problem, I deployed it offline, found it access this url, my server cannot access it:
https://fonts.googleapis.com/css2?family=Shrikhand&display=swap

**Comment by Lissy93** [comment]
@denganliang - Apologies, that appears to have been a mistake on my part. I've removed it in 330d812bfd868c01fd88e564e82232b9256ff5c0.

**Comment by lissy93** [comment]
This one is now done

The only features which require internet access are:
- Those you manually enable, see: [external requests](https://dashy.to/docs/privacy#external-requests)
- Certain icons - see #435
